yt_dlp/postprocessor/__init__.py

```diff
@@ -33,15 +33,38 @@
 from .sponskrub import SponSkrubPP
 from .sponsorblock import SponsorBlockPP
 from .xattrpp import XAttrMetadataPP
-from ..plugins import load_plugins
+from ..globals import plugin_pps, postprocessors
+from ..plugins import PACKAGE_NAME, register_plugin_spec, PluginSpec
+from ..utils import deprecation_warning
 
-_PLUGIN_CLASSES = load_plugins('postprocessor', 'PP')
+
+def __getattr__(name):
+    lookup = plugin_pps.value
+    if name in lookup:
+        deprecation_warning(
+            f'Importing a plugin Post-Processor from {__name__} is deprecated. '
+            f'Please import {PACKAGE_NAME}.postprocessor.{name} instead.')
+        return lookup[name]
+
+    raise AttributeError(f'module {__name__!r} has no attribute {name!r}')
 
 
 def get_postprocessor(key):
-    return globals()[key + 'PP']
+    return postprocessors.value[key + 'PP']
+
+
+register_plugin_spec(PluginSpec(
+    module_name='postprocessor',
+    suffix='PP',
+    destination=postprocessors,
+    plugin_destination=plugin_pps,
+))
 
+_default_pps = {
+    name: value
+    for name, value in globals().items()
+    if name.endswith('PP') or name in ('FFmpegPostProcessor', 'PostProcessor')
+}
+postprocessors.value.update(_default_pps)
```

Comment:
this will be lost after a `reset_plugins()` right?

Reply:
there is no reset_plugins()?

Reply:
tests...

Reply:
I don't think it clears postprocessors? load_plugins() only appends

the test does not clear postprocessors as far as I can tell

Reply:
I dont think it matters for tests as of now, but importing implicitly changes `plugin_pps` and `reset_plugins` does `plugin_pps.value = {}`, so maybe just something to keep in mind

Reply:
`reset_plugins()` in the test is hacky to account for lack of pytest isolation... but doesn't break things for testing as far as I can tell